android/tangram/src/com/mapzen/tangram/Marker.java

```diff
@@ -0,0 +1,189 @@
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.util.DisplayMetrics;
+
+import com.mapzen.tangram.geometry.Polygon;
+import com.mapzen.tangram.geometry.Polyline;
+
+/**
+ * Class used to display points, polylines, and bitmaps dynamically on a map. Do not create one of
+ * these objects directly, instead use {@link MapController#addMarker()}.
+ */
+public class Marker {
+
+    private Context context;
+    private long markerId = 0;
+    private MapController map;
+    private boolean visible = true;
+
+    /**
+     * Package private constructor for creating a new {@link Marker}.
+     * @param context the context to use for decoding bitmap resources
+     * @param markerId the marker id
+     * @param map the map this marker is added to
+     */
+    Marker(Context context, long markerId, MapController map) {
+        this.context = context;
+        this.markerId = markerId;
+        this.map = map;
+    }
+
+    /**
+     * Returns the {@link Marker} id
+     * @return marker id
+     */
+    public long getMarkerId() {
+        return markerId;
+    }
+
+    /**
+     * Sets the styling to be used to display either a point, polyline, or bitmap for this marker.
+     * If the marker is going to be used to display a bitmap, a 'points' style must be set.
+     *
+     * <ul>
+     * <li>{ style: 'points', color: 'white', size: [50px, 50px], order: 2000, collide: false }</li>
+     * <li>{ style: 'lines', color: '#06a6d4', width: 5px, order: 2000 }</li>
+     * <li>{ style: 'polygons', color: '#06a6d4', width: 5px, order: 2000 }</li>
+     * </ul>
+     *
+     * @param styleStr the style string
+     * @return whether the style was successfully set
+     */
+    public boolean setStyling(String styleStr) {
+        return map.setMarkerStyling(markerId, styleStr);
+    }
+
+    /**
+     * Sets the drawable resource id to be used to load a bitmap. When displaying a drawable, a
+     * 'points' style must also be set on the marker (see {@link Marker#setStyling(String)}.
+     *
+     * @param drawableId the drawable resource id
+     * @return whether the drawable's bitmap was successfully set
+     */
+    public boolean setDrawable(int drawableId) {
+        BitmapFactory.Options options = new BitmapFactory.Options();
+        options.inTargetDensity = DisplayMetrics.DENSITY_DEFAULT;
+        Bitmap bitmap = BitmapFactory.decodeResource(context.getResources(), drawableId, options);
+        return setBitmap(bitmap);
+    }
+
+    /**
+     * Sets the drawable to be used to load a bitmap. When displaying a drawable, a
+     * 'points' style must also be set on the marker (see {@link Marker#setStyling(String)}.
+     *
+     * @param drawable the drawable
+     * @return whether the drawable's bitmap was successfully set
+     */
+    public boolean setDrawable(Drawable drawable) {
```

Comment:
It might be nice to also have a public method exposed to set a raw bitmap. Raw bitmaps would be useful mostly for downloaded images where as drawables lend themselves to working with bundled resources.

This would be mostly for convenience so the client app would not need to convert the `Bitmap` to a `BitmapDrawable` just to have Tangram convert it back to a `Bitmap`. But maybe there is also an advantage to forcing developers to convert it to a `BitmapDrawable` so we can properly set target density? Although this should happen automatically [if the proper constructor is used](https://developer.android.com/reference/android/graphics/drawable/BitmapDrawable.html#BitmapDrawable(android.content.res.Resources, android.graphics.Bitmap)).